CompileSharpmake.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+# Script arguments:
+# $1: Project/Solution to build
+# $2: Target(Normally should be Debug or Release)
+# $3: Platform(Normally should be "Any CPU" for sln and AnyCPU for a csproj)
+# if none are passed, defaults to building Sharpmake.sln in Debug|Any CPU
+
+function BuildSharpmake {
+    solutionPath=$1
+    configuration=$2
+    platform=$3
+    echo Compiling $solutionPath in "${configuration}|${platform}"...
+    MSBUILD_CMD="msbuild -t:build -restore \"${solutionPath}\" /nologo /v:m /p:Configuration=${configuration} /p:Platform=\"${platform}\""
+    echo $MSBUILD_CMD
+    eval $MSBUILD_CMD
+    if [ $? -ne 0 ]; then
+        echo ERROR: Failed to compile $solutionPath in "${configuration}|${platform}".
+        exit 1
+    fi
+}
+
+# fail immediately if anything goes wrong
+set -e
+
+pushd $(dirname $0) > /dev/null
+CURRENT_DIR=$(pwd)
+popd > /dev/null
```

Comment:
why don't you directly store the result of dirname?

Reply:
Dirname give a relative path, wil pwd give a FullPath

Reply:
hum ok, on linux we have access to a cmd called `realpath` which does that, but it doesn't exist on mac for some reason :(